src/app/modules/main/chat_section/module.nim

```diff
@@ -388,11 +388,12 @@ proc rebuildCommunityTokenPermissionsModel(self: Module) =
 
 proc reevaluateRequiresTokenPermissionToJoin(self: Module) =
   let community = self.controller.getMyCommunity()
-  var hasBecomeMemberOrBecomeAdminPermissions = false
-  for id, tokenPermission in community.tokenPermissions:
-    if tokenPermission.`type` == TokenPermissionType.BecomeMember or tokenPermission.`type` == TokenPermissionType.BecomeAdmin:
-      hasBecomeMemberOrBecomeAdminPermissions = true
-  self.view.setRequiresTokenPermissionToJoin(hasBecomeMemberOrBecomeAdminPermissions)
+  var joinPermissionsChanged = false
+  for _, tokenPermission in community.tokenPermissions:
+    if tokenPermission.`type` == TokenPermissionType.BecomeMember or tokenPermission.`type` == TokenPermissionType.BecomeAdmin or tokenPermission.`type` == TokenPermissionType.BecomeTokenMaster:
```

Comment:
Check this out @endulab @jrainville pls

Reply:
@mprakhov too pls :)

Reply:
@mprakhov can you create/test that scenario? I was under the impression that `type == TokenPermissionType.BecomeTokenMaster` also qualifies as a "join" permission; if not, I can remove the condition here